patroni/validator.py

```diff
@@ -1058,6 +1058,7 @@ def validate_watchdog_mode(value: Any) -> None:
                     Optional("max_worker_processes"): IntValidator(0, 262143, raise_assert=True),
                 },
                 Optional("use_pg_rewind"): bool,
+                Optional("rewind"): [Or(str, dict)],
```

Comment:
You added it only to `bootstrap.dcs.postgresql`, but it also needs to be added to `postgresql` section.

Another thing, which is a bit related, we don't have validator for `basebackup` options, maybe you can add it.